geode-core/src/test/java/org/apache/geode/cache/client/internal/TXFailoverOpTest.java

```diff
@@ -0,0 +1,388 @@
+    }
+
+    @Override
+    public int getConnectionCount() {
+      return 0;
+    }
+
+    @Override
+    public Connection getConnection(Connection conn) {
+      return conn;
+    }
+
+    @Override
+    public void activate(Connection conn) {}
+
+    @Override
+    public void passivate(Connection conn, boolean accessed) {}
+  }
+
+  protected class DummyConnection implements Connection {
+
+    private ServerLocation server;
+
+    public DummyConnection(ServerLocation serverLocation) {
+      this.server = serverLocation;
+    }
+
+    @Override
+    public void close(boolean keepAlive) throws Exception {}
+
+    @Override
+    public void destroy() {
+      invalidateConnections++;
+    }
+
+    @Override
+    public boolean isDestroyed() {
+      return false;
+    }
+
+    @Override
+    public ByteBuffer getCommBuffer() {
+      return null;
+    }
+
+    @Override
+    public ServerLocation getServer() {
+      return server;
+    }
+
+    @Override
+    public Socket getSocket() {
+      return null;
+    }
+
+    @Override
+    public ConnectionStats getStats() {
+      return null;
+    }
+
+    @Override
+    public int getDistributedSystemId() {
+      return 0;
+    }
+
+    @Override
+    public Endpoint getEndpoint() {
+      return new Endpoint(null, null, null, null, null);
+    }
+
+    @Override
+    public ServerQueueStatus getQueueStatus() {
+      return null;
+    }
+
+    @Override
+    public Object execute(Op op) throws Exception {
+      throw new SocketTimeoutException();
+    }
+
+    @Override
+    public void emergencyClose() {}
+
+    @Override
+    public short getWanSiteVersion() {
+      return -1;
+    }
+
+    @Override
+    public void setWanSiteVersion(short wanSiteVersion) {}
+
+    @Override
+    public InputStream getInputStream() {
+      return null;
+    }
+
+    @Override
+    public OutputStream getOutputStream() {
+      return null;
+    }
+
+    @Override
+    public void setConnectionID(long id) {}
+
+    @Override
+    public long getConnectionID() {
+      return 0;
+    }
+  }
+
+  protected class DummyEndpointManager implements EndpointManager {
+
+    @Override
+    public void addListener(EndpointListener listener) {}
+
+    @Override
+    public void close() {}
+
+    @Override
+    public Endpoint referenceEndpoint(ServerLocation server, DistributedMember memberId) {
+      return null;
+    }
+
+    @Override
+    public Map getEndpointMap() {
+      return null;
+    }
+
+    @Override
+    public void removeListener(EndpointListener listener) {}
+
+    @Override
+    public void serverCrashed(Endpoint endpoint) {
+      serverCrashes++;
+    }
+
+    @Override
+    public int getConnectedServerCount() {
+      return 0;
+    }
+
+    @Override
+    public Map getAllStats() {
+      return null;
+    }
+
+    @Override
+    public String getPoolName() {
+      return null;
+    }
+  }
+
+  protected class DummyQueueManager implements QueueManager {
```

Comment:
Looks like you can mock this class: only the QueueConnections class implementation has any logic in it.